### PR TITLE
mcs: forward region requests to scheduling server

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5839.

### What is changed and how does it work?

This PR is used to forward region requests to the scheduling server and let it handle the scheduling directly, the operator will be brought through forwarded heartbeat response to API server. Then send it to TiKV.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by rleungx** [comment]
/test build
--

File: pkg/core/region.go [hunk 1]
```diff
@@ -194,6 +195,54 @@ func RegionFromHeartbeat(heartbeat *pdpb.RegionHeartbeatRequest, opts ...RegionC
 	return region
 }
 
+// RegionFromForward constructs a Region from forwarded region heartbeat.
+func RegionFromForward(heartbeat *schedulingpb.RegionHeartbeatRequest, opts ...RegionCreateOption) *RegionInfo {
```
Comment (lhy1024) [inline pkg/core/region.go:199]: RegionFromForward and RegionFromHeartbeat seem to be different only `regionKvSize := heartbeat.GetApproximateKvSize() / units.MiB` here.

File: pkg/mcs/scheduling/server/cluster.go [hunk 2]
```diff
@@ -348,3 +349,82 @@ func (c *Cluster) HandleStoreHeartbeat(heartbeat *schedulingpb.StoreHeartbeatReq
...
+	if err := c.processRegionHeartbeat(region); err != nil {
+		return err
+	}
+
+	c.coordinator.GetOperatorController().Dispatch(region, operator.DispatchFromHeartBeat, c.coordinator.RecordOpStepWithTTL)
+	return nil
+}
+
+// processRegionHeartbeat updates the region information.
+func (c *Cluster) processRegionHeartbeat(region *core.RegionInfo) error {
```
Comment (lhy1024) [inline pkg/mcs/scheduling/server/cluster.go:364]: Maybe we can reduce some of the duplicate functions, and of course, we can do it in the next pr.

File: server/cluster/cluster_test.go [hunk 3]
```diff
@@ -3633,7 +3633,7 @@ func TestInterval(t *testing.T) {
 func waitAddLearner(re *require.Assertions, stream mockhbstream.HeartbeatStream, region *core.RegionInfo, storeID uint64) *core.RegionInfo {
```
Comment (lhy1024) [inline server/cluster/cluster_test.go:3633]: Do we need to add some tests with API mode to test forward?

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

File: server/cluster/cluster_test.go (hunk 3 above)
Comment (rleungx) [reply server/cluster/cluster_test.go:3633]: added

File: pkg/core/region.go (hunk 1 above)
Comment (rleungx) [reply pkg/core/region.go:199]: request type is different

**Review by lhy1024 — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**